PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/thread_functions.cpp

```diff
@@ -0,0 +1,10 @@
+#include "thread_functions.h"
+
+void noramlFlowSender(Transmitter& transmitter, QueueWrapper<TLP*>& listenOn) {
+	while (true) {
+		// Wait until the queue is not empty
+		if (!listenOn.size()) {
+			transmitter.sendSequence(listenOn.popAll());
+		}
+	}
```

Comment:
popAll automatically blocks on the internal mutex when the queue is empty. No need to have it as a loop.